larray/io/excel.py

```diff
@@ -324,11 +324,17 @@ def shape(self):
             # see https://www.excelcampus.com/vba/find-last-row-column-cell/#endcode
             from xlwings.constants import Direction as xldir
             used = self.xw_sheet.api.UsedRange
-            last_row = max(used.Cells(used.Rows.Count + 1, col).End(xldir.xlUp).Row
-                           for col in range(1, used.Columns.Count+1))
-            last_col = max(used.Cells(row, used.Columns.Count + 1).End(xldir.xlToLeft).Column
-                           for row in range(1, used.Rows.Count+1))
-            return (last_row, last_col)
+            last_row = used.Row + used.Rows.Count
+            last_col = used.Column + used.Columns.Count
+            for last_used_row in range(last_row, 0, -1):
+                left_cell = used.Cells(last_used_row, last_col + 1).End(xldir.xlToLeft)
+                if left_cell.Column > 1 or (left_cell.Value is not None and left_cell.Value != ''):
```

Comment:
Are both cases possible? (ie can a cell be None or == '')?

Reply:
testing if Value is None seems to be enough.